pytoda/datasets/set_matching_dataset.py

```diff
@@ -0,0 +1,411 @@
+
+#             self.get_set_2_element = _get_set_2_element
+
+#     def set_crop_set_lengths(self) -> None:
+#         """
+#         Sets the function to crop the number of elements per set.
+#         """
+
+#         if not self.vary_set_length:
+#             self.crop_set_lengths = lambda set_1, set_2, idx: (set_1, set_2)
+#         else:
+#             # Set the seed (if applicable) and then randomly crop some elements
+#             # from one set and remove the correct ones from the other.
+#             def _crop_set_lengths(set_1, set_2, idx):
+#                 self.set_seed(self.seed + idx)
+#                 num_idxs = torch.randint(2, self.max_set_length + 1, (1,))
+#                 keep_idxs_1 = torch.randperm(self.max_set_length)[:num_idxs]
+
+#                 # This does exactly what x == y would do if x would be a Tensor and
+#                 # y a int, with the extension that y is an array.
+#                 keep_idxs_2 = torch.any(
+#                     torch.stack(
+#                         list(map(lambda x: x == self.permutation, keep_idxs_1))
+#                     ),
+#                     axis=0,
+#                 )
+#                 return set_1[keep_idxs_1, :], set_2[keep_idxs_2, :]
+
+#             self.crop_set_lengths = _crop_set_lengths
+
+#     def get_targets(self, set_1: torch.Tensor, set_2: torch.Tensor) -> Tuple:
+#         """Compute targets for one training sample
+
+#         Args:
+#             set_1 (torch.Tensor): Tensor with elements of set_1.
+#             set_2 (torch.Tensor): Tensor with elements of set_2.
+
+#         Returns:
+#             Tuple: Tuple containing hungarian matching indices of set1 vs set2 and
+#                 set2 vs set1.
+#         """
+
+#         cost_matrix = self.get_cost_matrix(set_1, set_2)
+
+#         matrix = torch.zeros_like(cost_matrix)
+#         rows, cols = linear_sum_assignment(cost_matrix)
+#         matrix[rows, cols] = 1
+#         idx_12 = torch.from_numpy(cols)
+#         idx_21 = torch.nonzero(matrix.t(), as_tuple=True)[1]
+
+#         return idx_12, idx_21
+
+#     def __len__(self) -> int:
+#         """Gets length of dataset.
+
+#         Returns:
+#             int: Length of the dataset being sampled.
+#         """
+#         return len(self.datasets[0])
+
+#     def __getitem__(self, index: int) -> Tuple:
+#         """Generates one sample from the dataset.
+
+#         Args:
+#             index (int): The index to be sampled.
+
+#         Returns:
+#             Tuple : Tuple containing sampled set1, sampled set2, hungarian
+#                 matching indices of set1 vs set2 and set2 vs set1.
+#         """
+
+#         set_1 = self.datasets[0][index]
+#         set_2 = self.get_set_2_element(set_1, index)
+#         set_1, set_2 = self.crop_set_lengths(set_1, set_2, index)
+#         targets_12, targets_21 = self.get_targets(set_1, set_2)
+#         return set_1, set_2, targets_12, targets_21
+
```

Comment:
Don't let old code sit around commented.

I know it can be convenient to compare while refactoring, it's noise in the history.
The old code is always there in version control if you need it.
So you can have code open in a browser on github, clone the repo checked out to older commit again under a different name and open in the IDE, or copy it in a temp file not added to the repo.

Reply:
ok, got it! sorry.